libs/transactions/units.ts

```diff
@@ -0,0 +1,57 @@
+import Web3 from "web3";
+
+function isString(s: any) {
+  return typeof s === "string" || s instanceof String;
+}
+
+export function toBaseUnit(value: any, decimals: number, web3: Web3) {
+  if (!isString(value)) {
+    throw new Error("Pass strings to prevent floating point precision issues.");
+  }
+  if (decimals === 0) {
```

Comment:
Can the `decimal` parameter be lower than zero? By reading the logic I think it can't be. We could add a validation for this.

Reply:
Sorry, I had to move on quickly so just copy+pasted this function. I'll make sure to review this logic as part of Issue #28 - Refactor `User.makeDeposit`.